### ISSUE TITLE
Explorer: option to disable auto open of files when no editor active

### ISSUE THREAD

**Issue by eyalroth** [microsoft/vscode#75181]
**Environment:**
- VSCode Version: 1.35.0
- OS Version: Windows 10.0.17134.799

**Steps to Reproduce:**
1. Disable setting _Explorer: Auto Reveal_.
2. Close settings.
3. Select a file in the explorer (but do not open it).
4. Open Keyboard Shortcuts window.
5. Trigger either _View: Show Explorer_ action or _View: Focus into Side Bar_.

**Bug:**

Selected file from step 3 is now revealed in a new editor (it shouldn't).

**Does this issue occur when all extensions are disabled?** Yes

**Comment by isidorn** [comment]
The autoReveal is not related to this behavior. 
The behavior is "When explorer is focused if there is no real editor active open the file". More about this here https://github.com/microsoft/vscode/issues/67708

We do not have a setting to disable this. And auto reveal is not it. This is the descirption of the autoReveal setting
"Controls whether the explorer should automatically reveal and select files when opening them."
So basically the other direction.


Leaving this open for now as a feature reuqest and renaming

**Comment by eyalroth** [comment]
@isidorn My bad, I didn't mean the auto-reveal setting but rather the editor preview (which I disabled), and actually you're right even if no window is open (not even the shortcut window) then this will happen.

I really _really_ dislike the preview feature and don't want anything resembling it in the editor ([perhaps except in the case of the search view](https://github.com/microsoft/vscode/issues/60131)), which this behavior definitely does. Moreover, I have the _Open Mode_ setting set to double-click, which is just another way for me to disable any kind of auto / immediate opening of files in the editor.

**Comment by isidorn** [comment]
I do understand what you want. However there is currently no way to achieve this, thus leaving this feature request open for now. And we can discuss further if more users give similar feedback.

I would not change this behavior out of the box since a lot of users are accustomed to it. And to introduce a setting I would not yet do.

**Comment by waynebloss** [comment]
This is really annoying to me as well. I close all the files to cleanup my workspace and then I press `Ctrl+Shift+E` to show the Explorer and whatever random file was highlighted previously is re-opened.

**Comment by waynebloss** [comment]
The logic presented in #67708 doesn't make any sense to me. It says:

> I like this flow because typically I open the SCM view to review changes and then when I am done I click back on the explorer to continue my work. Then it would automatically open that last file I was working on. I think we should get this behaviour back.

Why not just leave the files open that you were working? What if you had 2 files open? The `focus on explorer to open the previously highlighted file` only lets you get back to 1 file. Do people only work on one file at a time and not use tabs?

**Comment by waynebloss** [comment]
Why is this out of scope?

It's a very bad UX issue. I close all of my files and switching back to Explorer opens a random file that had previously been selected.

**Comment by telb99** [comment]
I would like this changed also.
Very frustrating to have files opened just because I switch to explorer from some other view after I close all my open file tabs.

**Comment by waynebloss** [comment]
I changed the keybinding of `Ctrl+Shift+E` from `workbench.view.explorer` to `workbench.files.action.collapseExplorerFolders`.

It's an imperfect solution because it collapses all the folders, but it also happens to show the explorer without opening any files... Not a solution for mouse users really.

I wonder if an extension could be built to fix this glaring UX issue. If anyone knows if the hooks are there, I'd certainly take a crack at it. My idea would be to add a new sidebar button, but instead of showing a custom panel, just show the explorer panel. Then I could hide the real explorer sidebar button or drag it all the way to the bottom. Also I'd create a new command to just show the explorer panel without opening any windows...

**Comment by javierrey** [comment]
Just joining in. Please prevent files from opening as a side effect of another intended action, i.e. moving or copying a file inside the explorer. It's useless, inefficient and confusing.

**Comment by aalok-sathe** [comment]
This is also a reason the File explorer cannot be used as a file manager, because deleting/renaming/performing any standard filesystem operation on a file inevitably opens a preview of the file

**Comment by mtbrolly** [comment]
This is especially annoying when the file you want to delete/rename/etc. is a file that is difficult to open in the editor (thinking of .npy files).

**Comment by robozb** [comment]
+1

**Comment by Vallek** [comment]
+1

**Comment by paolobiavati** [comment]
please add this setting

**Comment by shamsch** [comment]
oh god, will this now make me switch to neovim
